python/GafferSceneUI/ShaderQueryUI.py

```diff
@@ -0,0 +1,426 @@
+			> Note : If the location does not exist then the query will not be
+			> performed and all outputs will be set to their default values.
+			""",
+
+			"plugValueWidget:type", "GafferSceneUI.ScenePathPlugValueWidget",
+			"scenePathPlugValueWidget:scene", "scene",
+			"nodule:type", "",
+
+		],
+
+		"shader" : [
+
+			"description",
+			"""
+			The name of the shader to query.
+			> Note : If the shader does not exist then the query will not be
+			> performed and all outputs will be set to their default values.
+			""",
+
+			"plugValueWidget:type", "GafferUI.PresetsPlugValueWidget",
+			"presetsPlugValueWidget:allowCustom", True,
+
+			"preset:None", "",
+			"nodule:type", "",
+
+		],
+
+		"inherit" : [
+
+			"description",
+			"""
+			Should inherited shader attributes or be considered or not.
+			""",
+
+			"nodule:type", "",
+
+		],
+
+		"queries" : [
+
+			"description",
+			"""
+			The shader parameters to be queried - arbitrary numbers of shader
+			parameters may be added as children of this plug via the user interface,
+			or via python. Each child is a `NameValuePlug` whose `name` plug is
+			the shader parameter to query, and whose `value` plug is the default
+			value to use if the shader parameter can not be retrieved.
+
+			The full network of the shader given by `shader` is available
+			to be queried. Parameters on shaders in the network other than the
+			output shader can be specified as `shaderName.parameterName`.
+
+			> Note : If either the shader or parameter does not exist then the
+			> query will not be performed and all outputs will be set to their
+			> default values.
+			""",
+
+			"plugValueWidget:type", "GafferUI.LayoutPlugValueWidget",
+
+			"layout:customWidget:footer:widgetType", "GafferSceneUI.ShaderQueryUI._ShaderQueryFooter",
+			"layout:customWidget:footer:index", -1,
+
+			"nodule:type", "GafferUI::CompoundNodule",
+
+			"noduleLayout:section", "left",
+			"noduleLayout:spacing", 0.2,
+
+		],
+
+		"queries.*" : [
+
+			"nodule:type", "GafferUI::CompoundNodule",
+
+		],
+
+		"queries..." : [
+
+			"noduleLayout:section", "left",
+			"noduleLayout:label", functools.partial( __getLabel, parentPlug = "queries" ),
+
+			"description",
+			"""
+			The default value to use for the parameter.
+			""",
+
+			"spreadsheet:columnName", functools.partial( __getLabel, parentPlug = "queries", sanitise = True ),
+
+		],
+
+		"queries.*.name" : [
+
+			"description",
+			"""
+			The name of the parameter to query.
+			""",
+
+			"nodule:type", "",
+
+		],
+
+		"exists" : [
+
+			"description",
+			"""
+			Child plugs output true if the shader, location and parameter
+			exist, otherwise false. The order of the child plugs correspond to
+			the order of the query plugs.
+			""",
+
+			"plugValueWidget:type", "GafferUI.LayoutPlugValueWidget",
+
+			"layout:section", "Settings.Exists",
+
+			"nodule:type", "GafferUI::CompoundNodule",
+
+			"noduleLayout:section", "right",
+			"noduleLayout:spacing", 0.2,
```

Comment:
Exactly, especially if you start to get quite a few queries on one node, it will get taller as you go rather than wider. I feel like that's desirable, and it reinforces the shader-ness of the node. But I'm not married to it.

Reply:
It surprised me at first, but I think just because I'm used to the top-to-bottom flow of the other query nodes. It does seem plausible that matching the direction of shader outputs is going to work out better here...